apps/OpenSpace/ext/launcher/src/sgctedit/orientationdialog.cpp

```diff
@@ -0,0 +1,101 @@
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "sgctedit/display.h"
+#include "sgctedit/orientationdialog.h"
+
+OrientationDialog::OrientationDialog(sgct::quat& orientation, QWidget* parent)
+    : QDialog(parent)
+    , _orientationValue(orientation)
+{
+    setWindowTitle("Global Orientation");
+    QVBoxLayout* layoutWindow = new QVBoxLayout(this);
+
+    _linePitch = new QLineEdit(this);
+    _lineRoll = new QLineEdit(this);
+    _lineYaw = new QLineEdit(this);
+    _linePitch->setText(QString::number(_orientationValue.x));
+    _lineRoll->setText(QString::number(_orientationValue.z));
+    _lineYaw->setText(QString::number(_orientationValue.y));
+    {
+        QDoubleValidator* validatorPitch = new QDoubleValidator(-90.0, 90.0, 15);
+        QDoubleValidator* validatorRoll = new QDoubleValidator(-360.0, 360.0, 15);
+        QDoubleValidator* validatorYaw = new QDoubleValidator(-180.0, 180.0, 15);
+        validatorPitch->setNotation(QDoubleValidator::StandardNotation);
+        validatorRoll->setNotation(QDoubleValidator::StandardNotation);
+        validatorYaw->setNotation(QDoubleValidator::StandardNotation);
+        _linePitch->setValidator(validatorPitch);
+        _lineRoll->setValidator(validatorRoll);
+        _lineYaw->setValidator(validatorYaw);
+    }
+    {
+        QLabel* labelPitch = new QLabel(this);
```

Comment:
same here, n o need for explicitly passing this `this` pointer since they get reparented